### ISSUE TITLE
Accessibility Keyboard Interaction

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly-react#3108]
### Ultimate goals: 
- To have consistent keyboard behavior across similar components
- Have keyboard behavior documented for consumers/contributors
- Avoid regressions

### Updating Components:
- Component behaviors should be based off of [Design Documentation](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.1b322xkjq2i7)
- Current state of components has been audited, current behaviors are [documented](https://docs.google.com/spreadsheets/d/13PoYRnLhkiANYrkUxI1ekYVQVTp9Ity9CIBRpg5bKt0/edit#gid=872222636) and need updated based on best practices
- Each time we fix an interaction, a test should be included to prevent a regression
- Component behaviors that could use improvement are listed below

### Testing: 
- Each time we fix an interaction, a test should be included to prevent a regression
- Test framework (Cypress) has already been set up; it’s a matter of enabling tests to run and reducing the amount of time it takes to test. 
- [Design Documentation](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.1b322xkjq2i7) can be used as a blueprint for creating the tests
- Also investigate testing with Firefox/Chrome (and possibly other browsers)- different browsers tend to have different behaviors

### Component Behaviors: 
This section captures any keyboard behavior within the component that doesn’t currently meet accessibility expectations
** *NOTE:* ** Any issue that is created for fixing the components below should include creating tests as part of the acceptance criteria

#### [Dropdown:](https://github.com/patternfly/patternfly-react/issues/3210)
* [1.2 Toggle - Expanded state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.dm034j42xqbc)
	- Space
	- Esc
* [2.1 Panel - Escape user action](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.floi6nqyx5wy)
	- Esc
* [2.3 Panel > Menu - Single-selection variant](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.350ao5ffuycf)
	- Enter
	- Space
* [2.5 Panel > Menu > Menu Item - Managing focus](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.1lr7m0rc8a6s)
	- Tab
	- Shift + Tab

#### Dropdown Panel:
* [1.1 Toggle - Collapsed state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.43qwq5woonhu)
	- Enter
	- Space
* [1.2 Toggle - Expanded state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.dm034j42xqbc)
	- Space
	- Esc
* [2.1 Panel - Escape user action](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.floi6nqyx5wy)
	- Esc
* [2.2 Panel - Customizable variant](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#bookmark=id.n92a5yby57rr)
	- Tab
	- Shift + Tab

#### [Application Launcher](https://github.com/patternfly/patternfly-react/issues/3211) (uses Dropdown and will likely be impacted by Dropdown updates):
* [1.2 Toggle - Expanded state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.dm034j42xqbc)
	- Space (issue in Firefox)
	- Tab
	- Shift + Tab
* [2.1 Panel - Escape user action](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.floi6nqyx5wy)
	- Esc
* [Panel > Menu - Single-selection variant:](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.350ao5ffuycf)
	- Enter (Note that the behavior is roughly the same, except additional behavior is included in the example (e.g. the link places focus on the first heading, the other actions have additional behavior associated with them)
	- Space

#### [Context Selector:](https://github.com/patternfly/patternfly-react/issues/3205)
* [1.2 Toggle - Expanded state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.dm034j42xqbc)
	- Enter (issue in Firefox)
	- Esc
	- Tab
	- Shift + Tab
* [2.1 Panel - Escape user action](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.floi6nqyx5wy)
	- Esc
* [2.7 Panel > Menu - Filterable variant](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.9ma7faxlr9dt)
 	- Enter
 	- Space
 	- Tab
 	- Shift + Tab
 	- Arrow down
 	- Arrow up
* [2.8 Panel > Filter Input](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#bookmark=id.kyphqxq5cze4)
	- Shift + Tab

#### [Options Menu](https://github.com/patternfly/patternfly-react/issues/3213) (uses Dropdown and will likely be impacted by Dropdown updates):
* [1.2 Toggle - Expanded state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.dm034j42xqbc)
	- Space (issue in Firefox)
	- Esc
* [2.1 Panel - Escape user action](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.floi6nqyx5wy)
	- Esc
* [2.3 Panel > Menu - Single-selection variant](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#bookmark=id.qpnmzw86p8je)
	- Enter
	- Space
* [2.5 Panel > Menu > Menu Item - Managing focus](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.1lr7m0rc8a6s)
	- Tab
	- Shift + Tab

#### [Pagination](https://github.com/patternfly/patternfly-react/issues/3212) (uses Dropdown and will likely be impacted by Dropdown updates):
* [1.1 Toggle - Collapsed state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.43qwq5woonhu)
	- Space (issue in Firefox)
* [1.2 Toggle - Expanded state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.dm034j42xqbc)
	- Space (issue in Firefox)
	- Esc
* [2.1 Panel - Escape user action](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.floi6nqyx5wy)
	- Esc
* [2.3 Panel > Menu - Single-selection variant](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.350ao5ffuycf)
	- Enter (focus not placed)
	- Space
* [2.5 Panel > Menu > Menu Item - Managing focus](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.1lr7m0rc8a6s)
	- Tab
	- Shift + Tab

#### [Select:](https://github.com/patternfly/patternfly-react/issues/3206)
* [ 1.1 Toggle - Collapsed state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.43qwq5woonhu)
	- Space (issue in Firefox)
* [1.2 Toggle - Expanded state](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.dm034j42xqbc)
	- Space
	- Esc
	- Tab
	- Shift + Tab
* [2.1 Panel - Escape user action](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#heading=h.floi6nqyx5wy)
	- Esc
* [2.3 Panel > Menu - Single-selection variant](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#bookmark=id.qpnmzw86p8je)
	- Enter
	- Space
* [2.4 Panel > Menu - Multi-selection variant](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#bookmark=id.w6ckgyt5h1wg)
	- Space (issue with Chrome, but not FF. Check all browsers)
* [2.5 Panel > Menu > Menu Item - Managing focus](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#bookmark=id.66nmzdpbbejz)
 	- Tab
 	- Shift + Tab
* [2.6 Panel > Menu > Checkbox Menu Item - Managing focus](https://docs.google.com/document/d/1m9nICFRgCXnXeHYN487BMX9MHw4DjvOln2B5p8yJa70/edit#bookmark=id.1u3som1d6v7f)
 	- Shift + Tab
	- Arrow down (if next item is disabled)
	- Arrow up (if previous item is disabled)

**Comment by nicolethoen** [comment]
We will be refactoring the components to consume the `Menu` component to standardize interactions across similar components. Finishing the refactor process will be a breaking change.

**Comment by nicolethoen** [comment]
This can be considered a duplicate of https://github.com/patternfly/patternfly-react/issues/6014 since all dropdown menus will use the same Menu and/or Panel components under the hood, they will be using the same interaction patterns.
